react/features/app/actions.js

```diff
@@ -30,37 +30,33 @@ export function appNavigate(uri: ?string) {
  * state.
  * @param {Object} newLocation - The location URI to navigate to. The value
  * cannot be undefined and is assumed to have all properties such as
- * {@code host} and {@code room} defined values.
+ * {@code host} and {@code room} defined. The values for these properties, may
+ * be undefined.
  * @private
  * @returns {void}
  */
 function _appNavigateToMandatoryLocation(
         dispatch: Dispatch<*>, getState: Function,
         newLocation: Object) {
-    // TODO Kostiantyn Tsaregradskyi: We should probably detect if user is
-    // currently in a conference and ask her if she wants to close the
-    // current conference and start a new one with the new room name or
-    // domain.
-
-    const oldLocationURL = getState()['features/base/connection'].locationURL;
-    const oldHost = oldLocationURL ? oldLocationURL.host : undefined;
-    const newHost = newLocation.host;
-
-    if (oldHost === newHost) {
-        dispatchSetLocationURL()
-            .then(dispatchSetRoom);
-    } else {
-        // If the host has changed, we need to load the config of the new host
-        // and set it, and only after that we can navigate to a different route.
-        _loadConfig(newLocation)
-            .then(
-                config => configLoaded(/* err */ undefined, config),
-                err => configLoaded(err, /* config */ undefined))
-            .then(dispatchSetRoom);
-    }
+    let loadConfigPromise;
+
+    // Here we 'fake' a configuration when the room is undefined. We cannot
+    // load a configuration from the server, but we need lib-jitsi-meet to be
+    // initialized in order to be able to create some local tracks for the
+    // welcome page.
+
+    typeof newLocation.room === 'undefined'
```

Comment:
why not do this check inside `_loadConfig`? seems like a reasonable case for it to handle, and then `loadConfigPromise` could be made const and this code would just be:

`const loadConfigPromise = _loadConfig(newLocation)`

or, maybe do the ternary operator assignment a more traditional way (would also allow loadConfigPromise to be const):

`const loadConfigPromise = typeof newLocation.room === 'undefined' ? Promise.resolve() : _loadConfig(newLocation);`

Reply:
actually, if we do it the first way, we don't even the `loadConfigPromise` variable at all, and can just do:

`_loadConfig(newLocation).then(...)`

Reply:
Thanks for the suggestions! Sounds good, I'll make some adjustments.